models/migrations/v148.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func purgeInvalidDependenciesComments(x *xorm.Engine) error {
+	_, err := x.Exec("DELETE FROM comment WHERE dependent_issue_id NOT IN (SELECT id FROM issue)")
```

Comment:
I'm not a fan of this query for a few reasons
1, raw sql, when we could probably leverage xorm
2, IIRC sqlite may fail if `select id from issue` too large due to "too many arguments"
3, default value of dependant_issue_id is 0, and this could end up deleting many comments as issue_id 0 doesn't exist

Perhaps this could be broken up into batches of queries of issues where dependant_issue_id != 0, this would increase the number of queries run, but it likely would lead to a safer migration

Reply:
we may want to have a look at this too:
https://github.com/go-gitea/gitea/blob/master/models/migrations/v129.go

Reply:
I think the SQLite too many arguments issue is not the case here. That occurs if you do the selects in two stages and try to use `"... IN (?)", []string{....})`

Probably could use xorm here though although it would require two fake struct definitions.

Reply:
Good catch on the 0 dependent_issue_id though!

Reply:
think 1. & 2. can be considered as "resolved"

to 3. I'm happy if somebody can translate this query into xorm (/-builder), I had a try and it didnt worked out jet